### ISSUE TITLE
Cannot set Layout.showLegend(false) due to a conditional inclusion in Layout.getContext()

### ISSUE THREAD

**Issue by peterbanda** [jtablesaw/tablesaw#561]
if (!showLegend == DEFAULT_SHOW_LEGEND) context.put("showlegend", showLegend);

**Comment by lwhite1** [comment]
It seems i had the wrong default. it should be fixed now on master.

**Comment by peterbanda** [comment]
Thanks for checking/fixing this!

Sorry to reopen but I've played with it a bit more and for scatters (the chart type I've used/tested) I can confirm the legend is hidden by default for a single trace but shown by default for multiple ones, i.e., it might be tricky to provide a consistent (conditional) default value.

I assume you want to detect if the caller changed/touched the `showLegend`, for which case you might consider changing the type to Boolean and setting the attribute initially to null (instead of false).

This lib is quite handy for me and if needed I can volunteer some time on its development.
